cpp/src/io/fst/lookup_tables.cuh

```diff
@@ -179,6 +179,72 @@ class SingleSymbolSmemLUT {
   }
 };
 
+/**
+ * @brief A simple symbol group lookup wrapper that uses a simple function object to
+ * retrieve the symbol group id for a symbol.
+ *
+ * @tparam SymbolGroupLookupOpT The function object type to return the symbol group for a given
+ * symbol
+ */
+template <typename SymbolGroupLookupOpT>
+class SymbolGroupLookupOp {
+ private:
+  struct _TempStorage {};
```

Comment:
What is the purpose of this struct? It is empty, but as I know it may have size of 1 byte.

Reply:
This is part of the API of the finite-state transducer (FST) algorithm. You give it a triplet of:
1. `symbol group lookup table` that maps a symbol to 
2. `transition table` that maps a (`symbol_group_id`, `state`) to `new_state`
3. `translation table` that maps a (`symbol_group_id`, `state`) to a sequence of `symbols_to_output`

If any of these user-provided classes declares a `TempStorage` member type, the FST statically allocates shared memory for `TempStorage` and provides it to that class during initialization. This would allow the FST algorithm to "reclaim" that static shared memory allocation, when a class (e.g., `symbol group lookup table`) isn't needed in the algorithm anymore. This is a CUB-style implementation that I followed here. I don't think 1 byte should be a concern here, as we're still register-bound rather than shared memory-bound in the FST algorithm.

Reply:
That said, I have a refactor on the backburner to simplify the interface a bit by making this kind of wrapper superfluous for simple functor types.

Reply:
That said, I have a refactor on the backburner to simplify the interface a bit by making this kind of wrapper superfluous for simple functor types.